src/System.Private.DeveloperExperience.Console/src/Internal/DeveloperExperience/ConsolePal.Windows.cs

```diff
@@ -3,32 +3,40 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Text;
 
 namespace Internal.DeveloperExperience
 {
     internal static class ConsolePal
     {
-        private static IntPtr s_consoleErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
+        private static IntPtr s_stdErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
 
         internal unsafe static void WriteError(string errorMessage)
         {
-            if (s_consoleErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
+            if (s_stdErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
             {
                 // ensure we have a valid handle before writing to it
                 return; 
             }
 
-            fixed (char *pBuffer = errorMessage)
+            // Ensure that the encoding we use matches the current code page.
+            int currentCodePage = (int)Interop.mincore.GetConsoleOutputCP();
+            Encoding encoding;
+            try
             {
-                int numberOfCharsWritten;
-                Interop.mincore.WriteConsole(s_consoleErrorHandle, (byte*)pBuffer, errorMessage.Length, out numberOfCharsWritten, IntPtr.Zero);
+                encoding = Encoding.GetEncoding(currentCodePage);
```

Comment:
I think this component System.Private.DeveloperExperience can be used by the app when some crash happen. this can be used in actual UWP app. System.Console cannot be used in UWP because I believe it is using some APIs not allowed there. that is what I remember.

---
In reply to: [88738953](https://github.com/dotnet/corert/pull/2235#discussion_r88738953) [](ancestors = 88738953)